modules/exploits/multi/http/cockpit_cms_rce.rb

```diff
@@ -183,22 +183,23 @@ def report_cred(opts)
   def login(un, pass)
     print_status('Attempting login')
     res = send_request_cgi(
-      'uri' => '/auth/login'
+      'uri' => '/auth/login',
+      'keep_cookies' => true
     )
+
     fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to the web service") unless res
     fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to the web service") unless /csfr\s+:\s+"([^"]+)"/ =~ res.body
-    cookie = res.get_cookies
-    res = send_request_raw(
+
+    res = send_request_cgi(
       'uri' => '/auth/check',
       'method' => 'POST',
       'ctype' => 'application/json',
-      'cookie' => cookie,
       'data' => JSON.generate({ 'auth' => { 'user' => un, 'password' => pass }, 'csfr' => Regexp.last_match(1) })
     )
+    
     fail_with(Failure::UnexpectedReply, "#{peer} - Could not connect to the web service") unless res
     fail_with(Failure::UnexpectedReply, "#{peer} - Login failed. This is unexpected...") if res.body.include?('"success":false')
-    print_good("Valid cookie for #{un}: #{cookie}")
-    cookie
+    print_good("Valid cookie for #{un}: #{cookie_jar.cookies[0]}")
```

Comment:
I think previously `cookie` held _all_ cookies from the request this will just return the first one right?

Reply:
Yep 👍  good catch